selfservice/strategy/link/strategy_verification.go

```diff
@@ -323,8 +323,30 @@ func (s *Strategy) verificationUseToken(w http.ResponseWriter, r *http.Request,
 		return
 	}
 
-	http.Redirect(w, r, s.d.Config(r.Context()).SelfServiceFlowVerificationReturnTo(f.
-		AppendTo(s.d.Config(r.Context()).SelfServiceFlowVerificationUI())).String(), http.StatusFound)
+	returnTo, err := x.SecureRedirectTo(r, nil,
+		x.SecureRedirectAllowSelfServiceURLs(s.d.Config(r.Context()).SelfPublicURL(r)),
+		x.SecureRedirectAllowURLs(s.d.Config(r.Context()).SelfServiceBrowserWhitelistedReturnToDomains()),
+		x.SecureRedirectUseReturnToKey("after_verification"),
```

Comment:
This is doing the same thing as the code https://github.com/ory/kratos/pull/1149/files#diff-ff4686c1075c2eada24576348f772784fdac52ad16e7e52f707e0d192fbdbf34R340 but based on another query parameter. What does this solve?

Reply:
> Additionally, add support for passing an "after_verification" query parameter to flows which have a verification post hook (eg registration), which the user-agent will be redirected to after the post hook verification flow is completed.
Eg: initialize a registration flow with /self-service/registration/browser?return_to=[url1]&after_verification=[url2]
After completing the registration, the user will be redirected to url1, and a verification email will be sent to them.
After clicking the link in the verification email, they'll be redirected to url2.